RoaringBitmap/src/main/java/org/roaringbitmap/longlong/HighLowContainer.java

```diff
@@ -111,6 +108,56 @@ public boolean isEmpty() {
     return art.isEmpty();
   }
 
+  private void assertNonEmpty() {
+    if(isEmpty()) {
+      throw new NoSuchElementException("Empty " + this.getClass().getSimpleName());
+    }
+  }
+
+  private LeafNode getExtremeNode(Node root, boolean reverse) {
+    Node parent = root;
+    while (true) {
```

Comment:
Let's track depth instead and make this `while(depth < maxDepth)`. It should make no difference on a legal tree, but if someone accidentally builds some kind of illegal cyclic "tree", at least this'll error out, rather than getting stuck in an endless loop.

Reply:
Done. The change included moving this method into Art class